tests/unit/amo/components/TestCategory.js

```diff
@@ -75,163 +81,229 @@ describe('Category', () => {
     expect(root).toHaveClassName('Category');
   });
 
-  it('fetches categories when needed', () => {
-    const errorHandler = new ErrorHandler({
-      id: 'some-id',
-      dispatch: fakeDispatch,
+  it('should render an error', () => {
+    const root = render();
+    root.setProps({
+      errorHandler: createStubErrorHandler(
+        new Error('example of an error')
+      ),
     });
-    render({ category: null, errorHandler });
 
-    sinon.assert.calledWithMatch(fakeDispatch, _categoriesFetch({
-      errorHandlerId: 'some-id',
-    }));
+    expect(root.find(ErrorList)).toHaveLength(1);
   });
 
-  it('should return 404 if no category was found and loading is false', () => {
-    const root = mountRender({
-      category: null,
-      loading: false,
+  it('should render an error without a category too', () => {
+    const root = render({}, {
+      paramOverrides: { slug: 'unknown-category' },
+    });
+    root.setProps({
+      errorHandler: createStubErrorHandler(new Error('example')),
     });
 
-    expect(root.find(NotFound)).toHaveLength(1);
+    expect(root.find(ErrorList)).toHaveLength(1);
   });
 
-  it('should not return 404 if category is falsy and loading is true', () => {
-    const root = mountRender({ loading: true });
+  it('configures a Search component', () => {
+    const location = { query: { page: 2 } };
+    const params = {
+      slug: fakeCategory.slug,
+      visibleAddonType: visibleAddonType(fakeCategory.type),
+    };
+    const root = render({ location, params });
+    const search = root.find(Search);
 
-    expect(root.find(NotFound)).toHaveLength(0);
+    expect(search).toHaveProp('filters', {
+      addonType: fakeCategory.type,
+      category: fakeCategory.slug,
+      page: location.query.page,
+    });
+    expect(search).toHaveProp('paginationQueryParams', {
+      page: location.query.page,
+    });
+    expect(search).toHaveProp('pathname',
+      `/${params.visibleAddonType}/${fakeCategory.slug}/`);
+    expect(search).toHaveProp('enableSearchSort', false);
   });
 
-  it('should render an error', () => {
-    const errorHandler = createStubErrorHandler(
-      new Error('example of an error')
-    );
-    const root = render({ errorHandler });
+  it('fetches categories when not yet loaded', () => {
+    const fakeDispatch = sinon.stub(store, 'dispatch');
+    render({}, { autoDispatchCategories: false });
 
-    expect(root.find(ErrorList)).toHaveLength(1);
+    sinon.assert.calledWithMatch(fakeDispatch, categoriesFetch({
+      errorHandlerId: errorHandler.id,
+    }));
   });
 
-  it('should render an error without a category too', () => {
-    const errorHandler = createStubErrorHandler(
-      new Error('example of an error')
-    );
-    const root = render({ errorHandler, category: null, loading: false });
+  it('does not fetch categories when already loaded', () => {
```

Comment:
It would be fine because `!categories` will not be true for `[]` (an empty API response of categories). That would be a good test case to add though.